### PR TITLE
dist: Support SmartOS

### PR DESCRIPTION
**Pull request by geek**
This only includes the basic setup for now.  Still need to work on build

### CODE REVIEW

File: setup/smartos/ansible-playbook.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    - include_vars: ansible-vars.yaml
+      tags: vars
+
+    - name: General | pkgin Update
+      command: pkgin up
+      tags: general
+
+    - name: General | Install required packages
+      command: pkgin in {{ item }}
```
Comment (jbergstroem) [inline setup/smartos/ansible-playbook.yaml]: I had to add `-y`: `pkgin -y in {{ item }}`

File: setup/smartos/ansible-playbook.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      authorized_key: user="{{ server_user }}" key="{{ lookup('file', '/tmp/' + item + '.keys') }}"
+      with_items: ssh_users
+      tags: user
+
+    - name: Jenkins | Download Jenkins' slave.jar
+      command: curl -sL https://jenkins-iojs.nodesource.com/jnlpJars/slave.jar -o /home/{{ server_user }}/slave.jar
+      tags: jenkins
+
+    - name: Jenkins | Copy start.sh script
+      copy: src=./resources/start.sh dest=/home/{{ server_user }}/start.sh owner={{ server_user }} group={{ server_user }} mode=0755
```
Comment (jbergstroem) [inline setup/smartos/ansible-playbook.yaml]: This will fail if the group `iojs` is missing which doesn't seem to be added above.

Comment (geek) [inline setup/smartos/ansible-playbook.yaml]: Good catch, this is now fixed

**Comment by jbergstroem** [comment]
LGTM

**Comment by jbergstroem** [comment]
Another minor thing: Since path doesn't seem to be provided through `/etc/profile`we might want to add it as a envvar in the manifest (otherwise jenkins won't find path to git).

**Comment by jbergstroem** [comment]
(btw, confirmed that adding path fixed Jenkins)

**Comment by geek** [comment]
Closing in favor of #64

**Pull request closed without merging**